serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/service/virksert/VirkSertService.java

```diff
@@ -1,38 +1,51 @@
 package no.difi.meldingsutveksling.serviceregistry.service.virksert;
 
 import no.difi.meldingsutveksling.serviceregistry.config.ServiceregistryProperties;
-import no.difi.virksert.client.VirksertClient;
-import no.difi.virksert.client.VirksertClientBuilder;
-import no.difi.virksert.client.VirksertClientException;
+import no.difi.vefa.peppol.common.model.ParticipantIdentifier;
+import no.difi.vefa.peppol.common.model.ProcessIdentifier;
+import no.difi.vefa.peppol.common.model.Scheme;
+import no.difi.virksert.api.Mode;
+import no.difi.virksert.client.BusinessCertificateClient;
+import no.difi.virksert.client.lang.VirksertClientException;
+import no.difi.virksert.lang.BusinessCertificateException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
 import org.springframework.stereotype.Component;
 
 import javax.annotation.PostConstruct;
+import java.net.URISyntaxException;
 import java.security.cert.X509Certificate;
 
 @Component
 public class VirkSertService {
 
-    private VirksertClient virksertClient;
+    private BusinessCertificateClient virksertClient;
 
     @Autowired
     private ServiceregistryProperties properties;
 
+    @Autowired
+    private Environment env;
+
     public VirkSertService() {
     }
 
-    public VirkSertService(VirksertClient virksertClient) {
+    public VirkSertService(BusinessCertificateClient virksertClient) {
         this.virksertClient = virksertClient;
     }
 
     @PostConstruct
-    public void init() {
-        virksertClient = VirksertClientBuilder.newInstance()
-                .setUri(properties.getAr().getEndpointURL().toString()).build();
+    public void init() throws URISyntaxException, BusinessCertificateException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.net.URISyntaxException, no.difi.virksert.lang.BusinessCertificateException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)